modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/config/DefaultImplicitConfigurationProviderFactory.java

```diff
@@ -40,12 +42,14 @@ public ConfigurationProvider createImplicitConfigurationProvider(ExtensionModel
                                                                    Event event,
                                                                    MuleContext muleContext) {
     if (implicitConfigurationModel == null) {
-      throw new IllegalStateException(String.format(
-                                                    "Could not find a config for extension '%s' and none can be created automatically. Please define one",
-                                                    extensionModel.getName()));
+      throw new IllegalStateException("Could not find a config for extension '" + extensionModel.getName()
+          + "' and none can be created automatically. Please define one");
+    } else if (!canBeUsedImplicitly(implicitConfigurationModel)) {
+      throw new IllegalStateException("Configuration model '" + extensionModel.getName()
```

Comment:
From the end user POV it doesn't make sense to talk about implicit configuration models. The error should be the same as in line 45